src/server/controllers/match.controller.js

```diff
@@ -80,37 +138,52 @@ function arrayContainsElement(arr, val) {
 
 function checkCareerFilter(userInfo, filter) {
   if (userInfo === undefined) return false;
-  let userBool = {
-    area: filter.area === 'All' ? 1 : 0,
-    role: filter.role === 'All' ? 1 : 0,
-    years: filter.years === 'All' ? 1 : 0,
-    background: filter.education_background === 'All' ? 1 : 0,
-  };
-
-  if (userBool.area) {
-    userBool.role = 1;
-  } else {
-    userBool.area = userInfo.area === filter.area ? 1 : 0;
-    userBool.role = userInfo.role === filter.role || userBool.role ? 1 : 0;
-  }
-
-  if (userInfo.years === filter.years && !userBool.years) {
-    userBool.years = 1;
-  }
-
-  if (userInfo.education_background === filter.education_background && !userBool.background) {
-    userBool.background = 1;
-  }
-
-  if (userBool.area && userBool.role && userBool.years && userBool.background) return true;
-  else return false;
+  else return ((filter.area === 'All' || userInfo.area === filter.area)
```

Comment:
BTW, why do we have this "All" business? Shouldn't we just not have any filter if the filter value is "All"? It seems like a redundant check. Every filter should just be null by default (which implies All)